src/flb_log.c

```diff
@@ -57,10 +58,78 @@ static inline int consume_byte(flb_pipefd_t fd)
     return 0;
 }
 
+static inline bool log_file_sz_max(off_t size, struct flb_log *log)
+{
+    double sz_mb;
+
+    sz_mb = size / (1024.00 * 1024.00);
+    if (sz_mb >= log->max_sz_mb) {
+        return true;
+    }
+
+    return false;
+}
+
+static inline void log_file_rotate(struct flb_log *log)
+{
+    char *backup_file_name = NULL;
+    char *src_file_name = NULL;
+    int ret = -1;
+
+    if (log->max_history == 0) {
+        //no backups.
+        //remove the file since size has exceeded.
+        remove(log->out);
+        return;
+    }
+
+    //max 2 digits for backup name suffix and 1 each for '.' and '\0'.
+    backup_file_name = flb_malloc(strlen(log->out) + 4);
+    if (!backup_file_name) {
+        perror("malloc");
+        return;
+    }
+    src_file_name = flb_malloc(strlen(log->out) + 4);
```

Comment:
dont nest strlen, set the malloc size in a variable

Reply:
updated